pkg/preference/implem.go

```diff
@@ -253,32 +254,23 @@ func (c *preferenceInfo) SetConfiguration(parameter string, value string) error
 		switch p {
 
 		case "timeout":
-			typedval, err := strconv.Atoi(value)
+			typedval, err := time.ParseDuration(value)
 			if err != nil {
-				return fmt.Errorf("unable to set %q to %q", parameter, value)
-			}
-			if typedval < 0 {
-				return errors.New("cannot set timeout to less than 0")
+				return fmt.Errorf("unable to set %q to %q, value must be Duration (e.g. 4s, 5m, 1h)", parameter, value)
 			}
 			c.OdoSettings.Timeout = &typedval
 
 		case "pushtimeout":
-			typedval, err := strconv.Atoi(value)
+			typedval, err := time.ParseDuration(value)
 			if err != nil {
-				return fmt.Errorf("unable to set %q to %q, value must be an integer", parameter, value)
-			}
-			if typedval < 0 {
```

Comment:
Looks like `time.ParseDuration` does support all kinds of values.
```go
// ParseDuration parses a duration string.
// A duration string is a possibly signed sequence of
// decimal numbers, each with optional fraction and a unit suffix,
// such as "300ms", "-1.5h" or "2h45m".
// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
func ParseDuration(s string) (Duration, error) {
```

Reply:
> I was not aware of the --. 
>
Yes, it is a way to tell the CLI parser to stop further flag processing. Not sure, but I think this is a POSIX requirement, respected by most CLI parsers out there (at least the POSIX-compliant ones).

> I am surprised time.ParseDuration allows negative values,
>
Since a duration measures the difference between two time points, it can be negative if the end is before the start. To me, this is a valid case of allowing negative values.

Reply:
> Looks like `time.ParseDuration` does support all kinds of values.
> 
> ```go
> // ParseDuration parses a duration string.
> // A duration string is a possibly signed sequence of
> // decimal numbers, each with optional fraction and a unit suffix,
> // such as "300ms", "-1.5h" or "2h45m".
> // Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
> func ParseDuration(s string) (Duration, error) {
> ```

I don't know yet how we use all those Durations in `odo`, but maybe we could set a minimum acceptable value for each Duration. I don't think it makes sense to accept PushTimeout in microseconds for example :-) What do you think?

Reply:
What do you think about  `1 * time.Second` as a minimum acceptable value for preferences that expect time.Duration?
@feloy @rm3l

Reply:
> What do you think about `1 * time.Second` as a minimum acceptable value for preferences that expect time.Duration? 
>
That's fine to me. Setting a too-high min value would give less flexibility to users, I think.